src/tapedtask.jl

```diff
@@ -203,33 +200,57 @@ end
 copy_box(o, roster::Dict{UInt64, Any}) = o
 
 function Base.copy(x::Instruction, on_tape::Taped, roster::Dict{UInt64, Any})
+    # func may also be a boxed variable
+    func = copy_box(x.func, roster)
     input = map(x.input) do ob
         copy_box(ob, roster)
     end
     output = copy_box(x.output, roster)
-    Instruction(x.func, input, output, on_tape)
+    Instruction(func, input, output, on_tape)
+end
+
+function Base.copy(x::BlockInstruction, on_tape::Taped, roster::Dict{UInt64, Any})
+    args = map(x.args) do ob
+        copy_box(ob, roster)
+    end
+    BlockInstruction(x.block_id, args, on_tape)
 end
 
-function Base.copy(t::RawTape, on_tape::Taped, roster::Dict{UInt64, Any}; start::Int=1)
-    old_data = t
-    len = length(old_data) - start + 1
-    new_data = RawTape(undef, len)
+function Base.copy(x::BranchInstruction, on_tape::Taped, roster::Dict{UInt64, Any})
+    cond = copy_box(x.condition, roster)
+    args = map(x.args) do ob
+        copy_box(ob, roster)
+    end
+
+    BranchInstruction(cond, x.block_id, args, on_tape)
+end
+
+function Base.copy(x::ReturnInstruction, on_tape::Taped, roster::Dict{UInt64, Any})
+    arg = copy_box(x.arg, roster)
+    ReturnInstruction(arg, on_tape)
+end
 
-    for (i, x) in enumerate(old_data[start:end])
+function Base.copy(old_data::RawTape, on_tape::Taped, roster::Dict{UInt64, Any})
+    new_data = RawTape(undef, length(old_data))
```

Comment:
Maybe use `new_tape` and `old_tape` for better clarity? 
```suggestion
    new_tape = RawTape(undef, length(old_data))
```